packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,320 @@
+import {
+    attr,
+    FastElement,
+    nullableNumberConverter,
+    observable,
+    Observable,
+} from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+import {
+    Direction,
+    keyCodeArrowDown,
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+} from "@microsoft/fast-web-utilities";
+import { convertPixelToPercent } from "./slider-utilities";
+import { SliderConfiguration, SliderMode, SliderOrientation } from "./index";
+
+export class Slider extends FormAssociated<HTMLInputElement>
+    implements SliderConfiguration {
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+        this.setAttribute("aria-readonly", this.readOnly.toString());
+    }
+
+    public track: HTMLDivElement;
+    public thumb: HTMLDivElement;
+
+    @observable
+    public direction: Direction = Direction.ltr;
+
+    @observable
+    public isDragging: boolean = false;
+
+    @observable
+    public position: string;
+    @observable
+    public trackWidth: number = 0;
+    @observable
+    public trackMinWidth: number = 0;
+    @observable
+    public trackHeight: number = 0;
+    @observable
+    public trackMinHeight: number = 0;
+
+    /**
+     * The element's value to be included in form submission changed.
+     */
+    @attr
+    public value: string; // Map to proxy element.
```

Comment:
Should value have a converter to ensure that the property is always a number?

Reply:
done and done nullableNumberConverter used